test/test_constraints_file.py

```diff
@@ -50,18 +50,100 @@ def test_add_at_least_one_employee_per_shift_constraint():
+    assert (status == cp_model.OPTIMAL)
+    expected_employee_working = False
+
+    for shift in [test_shift1, test_shift2, test_shift3]:
+        for employee in [test_employee, test_employee2]:
+            working_assignment = all_shifts[FrozenShiftCombinationsKey(employee.employee_id, shift.shift_id)]
+            # the employee does not work this shift
+            assert (solver.Value(working_assignment) == expected_employee_working)
+
+
+# A test that checks if the constraint 'add_at_most_one_shift_in_the_same_day_constraint' ensures that an employee does 
+# not work more than 1 shift a day (at most one shift a day). When combining the 2 constraints 
+# (add_exactly_one_employee_per_shift_constraint) the model needs to assign exactly one employee in each shift and at 
+# least one shift per employee per day. When testing with only 2 employees and 3 shift, there is no optimal solution 
+# because there are fewer employees than the shift in the same day. 
+def test_add_at_most_one_shift_per_employee_in_the_same_day_constraint_with_at_least_one_employee_in_a_shift_with_3_shifts_2_employees():
+    test_employee = Employee("test", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+    test_employee2 = Employee("test2", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+    test_shift1 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.MORNING, start_time=datetime.datetime(2023, 12, 11, 9, 30), end_time=datetime.datetime(2023, 12, 11, 16, 0))
+    test_shift2 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.EVENING, start_time=datetime.datetime(2023, 12, 11, 16, 0), end_time=datetime.datetime(2023, 12, 11, 22, 0))
+    test_shift3 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.CLOSING, start_time=datetime.datetime(2023, 12, 11, 17, 30), end_time=datetime.datetime(2023, 12, 12, 2, 0))
+
+    shifts = [test_shift1, test_shift2, test_shift3]
+    employees = [test_employee, test_employee2]
+
+    model = cp_model.CpModel()
+    all_shifts = generate_shift_employee_combinations(employees, shifts, model)
+
+    # When adding the constraint "add_one_employee_per_shift_constraint", it also ensures that there is exactly one
+    # employee in each shift, causing "add_at_most_one_shift_in_the_same_day_constraint" to fail; because the solver
+    # needs to assign exactly one employee in each shift, and cannot assign an employee to at most 1 shift a day.
+    add_exactly_one_employee_per_shift_constraint(shifts, employees, model, all_shifts)
+    add_at_most_one_shift_per_employee_in_the_same_day_constraint(shifts, employees, model, all_shifts)
+
+    solver = cp_model.CpSolver()
+    status = solver.Solve(model)
+    assert (status != cp_model.OPTIMAL)
+
+
+def test_add_at_most_one_shift_per_employee_in_the_same_day_constraint_with_at_least_one_employee_in_a_shift():
```

Comment:
you arent saying what you are actually testing. what is the state you are testing